x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -917,6 +917,7 @@ private RangeMissingHandler writerWithOffset(RangeMissingHandler writer, CacheFi
                 return (channel, channelPos, relativePos, len, progressUpdater) -> {
                     assert assertValidRegionAndLength(fileRegion, channelPos, len);
                     adjustedWriter.fillCacheRange(channel, channelPos, relativePos, len, progressUpdater);
+                    assert regionOwners.get(fileRegion.io) == fileRegion;
```

Comment:
Maybe assert before filling the cache range?

Reply:
It's already asserted before it with `assert assertValidRegionAndLength(fileRegion, channelPos, len)`.